### PR TITLE
feat: upgrade to MDX v2

### PR DESCRIPTION
**Pull request by tthsympa**
## How to test
- Go on [this MR](https://github.com/scaleway/docs/pull/1480)
  - `nvm use && rm -rf ./node_modules && yarn && yarn clean`
  - `cd content && git checkout fix/mdxv2 && git pull origin fix/mdxv2`
  - `cd .. && yarn build`
- Make sure build is ok
- Check that pages are vastly ok
- Try to use the `--pages` flag. The flag let you now build complete subfolder or single page.
  - `yarn build --pages=compute` will build all the compute doc
  - `yarn build --pages=home` will build the homepage
  - `yarn build --pages=tutorials/xxxxxxx` will build only this specific page

## Wtf man?
Following the upgrade to MDX v2 on the source code, I did a pass on the syntax of your files. It's important to upgrade for the good health of the project!

### **The bad news**
The only change that has a negative impact is concerning bullet listing (`ol-li`). The good syntax is now to have two tabs below each `li`. If you don't put those 2 tabs, the ordering will break (so instead of having `1. / 2. / 3.`, you will see `1. / 1. / 1.`). I did a pretty big check on a lit of files but of course I might not fix all of them. You'll be able to fix it on the go (please ? 🙏)

#### Blocks
Content in the `blocks/` folder are now `.json` files (instead of `.mdx`).

#### Frontmatter
`image` is replaced by `hero`. You need to specify the relative file of the asset 
**Example:**
```
---
...
hero: assets/general.webp
---
```

#### Lightbox
You don't need to specify the `assets/` in the `src` props of the component. Also, the system behind how we handle images have changed: it is highly recommended to uniquely name images.
**Example:**
```jsx
<Lightbox src="scaleway_xxxx_xxxx.webp" />
```

#### `<` character
You can't a `<` alone like this on your MDX. If you really want to display the `<` character, you must escape it with `\`
```
This is free for \< 1 hour
```

#### bullet list
Always add two tabs (4 spaces) below each `li` of the list. Also, don't put new line between each `li`.

Exemple:
Instead of
```jsx
1. xxxxxx
  I'm a string with two spaces before

2. xxxxxxx
I'm a string with no space before
3. xxxxxxxxx
  <Message ...>
 I'm content of a message!
  </Message>
```
do
```jsx
1. xxxxxx
    I'm a string with 4 spaces before
2. xxxxxxx
    I'm also a string with 4 spaces before
3. xxxxxxxxx
    <Message ...>
      I'm content of a message!
    </Message>
```

#### Message
It's not necessary to add new line after/before JSX tag. Also, add one tab (two spaces) for the content to show it's nested in it (like you would do in HTML)

Example:
Instead of 
```jsx
<Message type="note">

- xxxxxx
- xxxxxx
- xxxxxx
- xxxxxx

</Message>
```
do
```jsx
<Message type="note">
  - xxxxxx
  - xxxxxx
  - xxxxxx
  - xxxxxx
</Message>
````

### CODE REVIEW

File: faq/blockstorage.mdx [hunk 1]
```diff
@@ -5,12 +5,12 @@ meta:
...
-image: blockstorage.webp
+hero: assets/blockstorage.webp
 ---
 
 ## Why should I use Block Storage instead of local storage?
 
-The size of local storage depends on your Instance type. With today’s configurations, the maximum local storage available is 600 GB.
+The size of local storage depends on your Instance type. With today's configurations, the maximum local storage available is 600 GB.
 
 Block storage allows you to create bigger volumes and those volumes are persistent which means that they can live without being attached to an Instance.
```
Comment (iManu) [inline faq/blockstorage.mdx:16]: Table is broken below

Capture d’écran 2022-10-19 à 9 37 23 AM

File: contribute.mdx [hunk 2]
```diff
@@ -35,106 +35,77 @@ Get paid for the content you write that we publish. You can get up to €200 dis
 ## How to get started
```
Comment (iManu) [inline contribute.mdx:35]: Hero image not showed (white square), here on local build:
Capture d’écran 2022-10-19 à 9 44 42 AM

It should be used with srcset stuffs, here in prod:

Capture d’écran 2022-10-19 à 9 45 12 AM

File: datacenters/datacenter-as-a-service/brochures.mdx [hunk 3]
```diff
@@ -19,7 +19,7 @@ Scaleway Datacenter teams guide you through all types of colocation projects (ho
 : Ideal for large private cages and private suites. 
 
 [Scaleway DC4 (PDF, ≈ 1MB)](https://datacenter.scaleway.com/pdf/scaleway-dc4.pdf)
-: Cloud-enabler hotel in Paris’ city center.
+: Cloud-enabler hotel in Paris' city center.
 
 [Scaleway DC5 (PDF, ≈ 1MB)](https://datacenter.scaleway.com/pdf/scaleway-dc5.pdf)
 : Built for large-scale colocation.
```
Comment (iManu) [inline datacenters/datacenter-as-a-service/brochures.mdx:25]: PDFs "Our strengths" have no links nor sizes
Capture d’écran 2022-10-19 à 9 49 52 AM

File: storage/block/how-to/create-a-volume.mdx [hunk 4]
```diff
@@ -11,27 +11,23 @@ dates:
...
-  </Message>
-
-  <Lightbox src="scaleway-choose-region.webp" alt="" />
+    <Lightbox src="scaleway-choose-region.webp" alt="" />
 5. Select Block Storage when configuring the volume. Enter the volume name and select a size.
-  <Lightbox src="scaleway-configure-block-volume.webp" alt="" />
+    <Lightbox src="scaleway-configure-block-volume.webp" alt="" />
 6. Click **Create a new volume** to confirm the creation of the volume.
 
 <Navigation title="See Also">
```
Comment (iManu) [inline storage/block/how-to/create-a-volume.mdx:33]: Navigation next button should be in right position.

Here it is in left at local build:
Capture d’écran 2022-10-19 à 10 20 47 AM

But in right at prod:

Capture d’écran 2022-10-19 à 10 20 52 AM

File: faq/dns.mdx [hunk 5]
```diff
@@ -5,7 +5,7 @@ meta:
 content:
   h1: DNS
   paragraph: Discover DNS.
-image: DNS.webp
```
Comment (iManu) [inline faq/dns.mdx:8]: Image missiing : update to uppercase OR change webp file name to lowercase

File: faq/vpc.mdx [hunk 6]
```diff
@@ -5,7 +5,7 @@ meta:
 content:
   h1: VPC
   paragraph: Discover Private Networks and Public Gateways.
-image: VPC.webp
```
[the remark this answers is not in the document]
Comment (iManu) [reply faq/vpc.mdx:8]: Same in related faq page

File: faq/dns.mdx (hunk 5 above)
Comment (iManu) [reply faq/dns.mdx:8]: Same in related faq page

File: storage/block/how-to/create-a-volume.mdx (hunk 4 above)
Comment (tthsympa) [reply storage/block/how-to/create-a-volume.mdx:33]: C'est ok chez moi en build local ici

**Review by RoRoJ — APPROVED** [review]
I built and tested, every page I tested seemed fine.

It's a bit annoying that we can't have line breaks between numbered steps any more, just for the readability of some of the long/dense pages when we are editing them. But I guess we'll get used to it!

We need to remember to update https://confluence.infra.online.net/display/MARKETTECHANIM/DOCUMENTATION+GUIDELINES#DOCUMENTATIONGUIDELINES-TechnicalGuidelines once these changes are merged

**Review by iManu — APPROVED** [review]

**Pull request merged**